### PR TITLE
Restrict project creation to staff users

### PR DESCRIPTION
**Pull request by youichiro**
## Issue
https://github.com/doccano/doccano/issues/935

## What this PR do
- restrict project creation to staff users.
- don't display create and delete project buttons if login user is not a staff.

## Why
- Annotator and approver don't need to create a project.
  - If necessary, a superuser changes the general user to staff.
- Issue #935 says to add a new user type, but I implement without changing DB schema for now.
  - Staff can login to the admin page, but they can't do anything unless a superuser grants them some permissions.
  - If you do not want to change a general user to staff, we will need to add a new (guest) user type and change the restrictions.

## Screenshot
Create and delete buttons are hidden when login user is not a staff.
スクリーンショット 2021-07-21 1 14 46

## Concern
If you already let general users creating projects, you will need to update that users information.

### CODE REVIEW

File: frontend/store/auth.js [hunk 1]
```diff
@@ -46,13 +53,16 @@ export const actions = {
       commit('setAuthenticated', true)
       commit('setUsername', user.username)
       commit('setUserId', user.id)
+      commit('setIsStaff', false)
```
Comment (Hironsan) [inline frontend/store/auth.js:56]: Is `user.isStaff` correct?
Even if the user is staff, the create button is hidden:

File: backend/api/views/project.py [hunk 2]
```diff
@@ -17,7 +18,10 @@ def get_queryset(self):
         return self.request.user.projects
 
     def perform_create(self, serializer):
-        serializer.save(users=[self.request.user])
+        if self.request.user.is_staff:
+            serializer.save(users=[self.request.user])
+        else:
+            raise ProjectCreationPermissionDenied()
```
Comment (Hironsan) [inline backend/api/views/project.py:24]: Maybe non-staff user but project admin can delete projects.
Defining staff permission and setting it to `permission_classes` may be a safer way.

File: frontend/store/auth.js (hunk 1 above)
Comment (youichiro) [reply frontend/store/auth.js]: fixed. https://github.com/doccano/doccano/pull/1453/commits/a6397eb5ec0071e544fd470a0eb778d17d8bfb3e

File: backend/api/views/project.py (hunk 2 above)
Comment (youichiro) [reply backend/api/views/project.py]: Thanks for your comments.
Changed to use `permission_classes`. https://github.com/doccano/doccano/pull/1453/commits/ed37696a67a0e1dc90fa5fdcd0c566e223a7b6c0

**Comment by Hironsan** [comment]
Merged. Thanks!

**Pull request merged**